web/packages/teleport/src/Console/DocumentKubeExec/KubeExecDataDialog.tsx

```diff
@@ -0,0 +1,171 @@
+  ButtonPrimary,
+  ButtonSecondary,
+  Flex,
+  Text,
+  Toggle,
+} from 'design';
+
+import Validation from 'shared/components/Validation';
+import FieldInput from 'shared/components/FieldInput';
+import { requiredField } from 'shared/components/Validation/rules';
+import { ToolTipInfo } from 'shared/components/ToolTip';
+
+type Props = {
+  onClose(): void;
+  onExec(
+    namespace: string,
+    pod: string,
+    container: string,
+    command: string,
+    isInteractive: boolean
+  ): void;
+};
+
+function KubeExecDataDialog({ onClose, onExec }: Props) {
+  const [namespace, setNamespace] = useState('');
+  const [pod, setPod] = useState('');
+  const [container, setContainer] = useState('');
+  const [execCommand, setExecCommand] = useState('');
+  const [execInteractive, setExecInteractive] = useState(true);
+
+  const podExec = () => {
+    onExec(namespace, pod, container, execCommand, execInteractive);
+  };
+
+  return (
+    <Dialog
+      dialogCss={dialogCss}
+      disableEscapeKeyDown={false}
+      onClose={onClose}
+      open={true}
+    >
+      <Validation>
+        {({ validator }) => (
+          <form>
+            <DialogHeader>
+              <DialogTitle>Exec into a pod</DialogTitle>
+            </DialogHeader>
+            <DialogContent>
+              <Box mb={4}>
+                <Flex gap={3}>
+                  <FieldInput
+                    value={namespace}
+                    placeholder="namespace"
+                    label="Namespace"
+                    autoFocus
+                    rule={requiredField('Namespace is required')}
+                    width="33%"
+                    onChange={e => setNamespace(e.target.value.trim())}
+                  />
+                  <FieldInput
+                    value={pod}
+                    placeholder="pod"
+                    label="Pod"
+                    rule={requiredField('Pod is required')}
+                    width="33%"
+                    onChange={e => setPod(e.target.value.trim())}
+                  />
+                  <FieldInput
+                    value={container}
+                    placeholder="container"
+                    label="Container (optional)"
+                    width="33%"
+                    onChange={e => setContainer(e.target.value.trim())}
+                  />
+                </Flex>
+                <Flex justifyContent="space-between" gap={3}>
+                  <FieldInput
+                    rule={requiredField('Command to execute is required')}
+                    value={execCommand}
+                    placeholder="/bin/bash"
+                    label="Command to execute"
+                    width="66%"
+                    onChange={e => setExecCommand(e.target.value)}
+                    toolTipContent={
+                      <Text>
+                        The command that will be executed inside the target pod.
+                      </Text>
+                    }
+                  />
```

Comment:
Yeah, that's a nodes list page, so currently it's only about SSH. I guess it would be part of expanding kubectl in webui work to add kube there, along with the possibility of joining session in the web ui.

Regarding length limit, it looks like we generally don't do that in the UI, we don't have length rule in our standard validation rule set. Kubernetes has limits on length of namespace/pod/container so we could enforce it, but I think we can just do that on the backend, returning something like `trace.BadParameter("Namespace is too long, should be less than 253 symbols")`, I see it as a very rare edge case and nobody with good intentions probably would hit it.

Reply:
Yeah, I meant that we only need to enforce the limit on the backend without fiddling with validation for that on the client side, since it can't be an honest mistake.

Reply:
Added validation on the backend: https://github.com/gravitational/teleport/pull/41466/commits/4d8974017f65e6c3588fb4b8d77ba760b1b9a2d3

Reply:
We can not close the document on graceful disconnect, then this UI won't show up.

Reply:
Yeah, but that UI is general, I mean, I think it should be there - you might want to open an ssh connection alongside the kube exec connection. But I think it should be ok for the first version? And once we expand UI support for kube in web, it will be more natural, where you will be able to choose which type of resource you want to connect to.